src/sagemaker_huggingface_inference_toolkit/transformers_utils.py

```diff
@@ -231,15 +232,17 @@ def infer_task_from_model_architecture(model_config_path: str, architecture_inde
         architecture = config.get("architectures", [None])[architecture_index]
 
     task = None
+    if os.environ["HF_TASK"] and os.environ["HF_TASK"] in TaskManager.get_all_tasks():
```

Comment:
do we need to include the optimum dependency? User should just be able to go ahead and try the task I think. 

If we use TaskManager, we could run into similar issues down the line as with ARCHITECTURES_2_TASK right? Say TaskManager has not updated with the newest tasks but get_pipeline() has. This could possibly happen because TaskManager is from Optimum and get_pipeline() is from Transformers. They may not be in sync

the get_pipeline() call in the inference.py will do all the validation for us as it is the one loading the model

Reply:
Looks like Optimum is an extension, so they should ideally not be out of sync. Valid concern though, I can deep dive and find the source of truth of Optimum's tasks set

From https://huggingface.co/docs/optimum/index
🤗 Optimum is an extension of [Transformers](https://huggingface.co/docs/transformers) that provides a set of performance optimization tools to train and run models on targeted hardware with maximum efficiency.

Reply:
Looks like Phillip responded as well, I will remove this check